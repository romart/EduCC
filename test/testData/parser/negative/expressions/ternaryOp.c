int i;
float f;
int *pi;
float *pf;

int arr[20];

struct S {};

struct S s, *ps;

void testTernary(int cond) {
  cond ? i : i;
  cond ? i : f;
  cond ? i : pi;
  cond ? i : pf;
  cond ? i : arr;
  cond ? i : s;
  cond ? i : ps;
  cond ? f : i;
  cond ? f : f;
  cond ? f : pi;
  cond ? f : pf;
  cond ? f : arr;
  cond ? f : s;
  cond ? f : ps;
  cond ? pi : i;
  cond ? pi : f;
  cond ? pi : pi;
  cond ? pi : pf;
  cond ? pi : arr;
  cond ? pi : s;
  cond ? pi : ps;
  cond ? pf : i;
  cond ? pf : f;
  cond ? pf : pi;
  cond ? pf : pf;
  cond ? pf : arr;
  cond ? pf : s;
  cond ? pf : ps;
  cond ? arr : i;
  cond ? arr : f;
  cond ? arr : pi;
  cond ? arr : pf;
  cond ? arr : arr;
  cond ? arr : s;
  cond ? arr : ps;
  cond ? s : i;
  cond ? s : f;
  cond ? s : pi;
  cond ? s : pf;
  cond ? s : arr;
  cond ? s : s;
  cond ? s : ps;
  cond ? ps : i;
  cond ? ps : f;
  cond ? ps : pi;
  cond ? ps : pf;
  cond ? ps : arr;
  cond ? ps : s;
  cond ? ps : ps;
}
