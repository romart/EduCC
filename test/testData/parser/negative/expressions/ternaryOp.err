test/testData/parser/negative/expressions/ternaryOp.c:10:18: error: incompatible operand types ('signed int' and 'STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:22: error: incompatible operand types ('float' and '*signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:23: error: incompatible operand types ('float' and '*float')
test/testData/parser/negative/expressions/ternaryOp.c:10:24: error: incompatible operand types ('float' and 'signed int[20]')
test/testData/parser/negative/expressions/ternaryOp.c:10:25: error: incompatible operand types ('float' and 'STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:26: error: incompatible operand types ('float' and '*STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:28: error: incompatible operand types ('*signed int' and 'float')
test/testData/parser/negative/expressions/ternaryOp.c:10:32: error: incompatible operand types ('*signed int' and 'STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:35: error: incompatible operand types ('*float' and 'float')
test/testData/parser/negative/expressions/ternaryOp.c:10:39: error: incompatible operand types ('*float' and 'STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:42: error: incompatible operand types ('signed int[20]' and 'float')
test/testData/parser/negative/expressions/ternaryOp.c:10:46: error: incompatible operand types ('signed int[20]' and 'STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:48: error: incompatible operand types ('STRUCT S' and 'signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:49: error: incompatible operand types ('STRUCT S' and 'float')
test/testData/parser/negative/expressions/ternaryOp.c:10:50: error: incompatible operand types ('STRUCT S' and '*signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:51: error: incompatible operand types ('STRUCT S' and '*float')
test/testData/parser/negative/expressions/ternaryOp.c:10:52: error: incompatible operand types ('STRUCT S' and 'signed int[20]')
test/testData/parser/negative/expressions/ternaryOp.c:10:54: error: incompatible operand types ('STRUCT S' and '*STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:56: error: incompatible operand types ('*STRUCT S' and 'float')
test/testData/parser/negative/expressions/ternaryOp.c:10:60: error: incompatible operand types ('*STRUCT S' and 'STRUCT S')
