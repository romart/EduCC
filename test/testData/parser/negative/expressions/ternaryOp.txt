FILE test/testData/parser/negative/expressions/ternaryOp.c
  STRUCT S
----
  signed int i
----
  float f
----
  *signed int pi
----
  *float pf
----
  signed int[20] arr
----
  STRUCT S s
----
  *STRUCT S ps
----
  FUN void testTernary 
    #0: signed int cond
  BEGIN
    *cond ? *i : *i
    *cond ? *i : *f
    *cond ? *i : *pi
    *cond ? *i : *pf
    *cond ? *i : *arr
    *cond ? *i : *s
    *cond ? *i : *ps
    *cond ? *f : *i
    *cond ? *f : *f
    *cond ? *f : *pi
    *cond ? *f : *pf
    *cond ? *f : *arr
    *cond ? *f : *s
    *cond ? *f : *ps
    *cond ? *pi : *i
    *cond ? *pi : *f
    *cond ? *pi : *pi
    *cond ? *pi : *pf
    *cond ? *pi : *arr
    *cond ? *pi : *s
    *cond ? *pi : *ps
    *cond ? *pf : *i
    *cond ? *pf : *f
    *cond ? *pf : *pi
    *cond ? *pf : *pf
    *cond ? *pf : *arr
    *cond ? *pf : *s
    *cond ? *pf : *ps
    *cond ? *arr : *i
    *cond ? *arr : *f
    *cond ? *arr : *pi
    *cond ? *arr : *pf
    *cond ? *arr : *arr
    *cond ? *arr : *s
    *cond ? *arr : *ps
    *cond ? *s : *i
    *cond ? *s : *f
    *cond ? *s : *pi
    *cond ? *s : *pf
    *cond ? *s : *arr
    *cond ? *s : *s
    *cond ? *s : *ps
    *cond ? *ps : *i
    *cond ? *ps : *f
    *cond ? *ps : *pi
    *cond ? *ps : *pf
    *cond ? *ps : *arr
    *cond ? *ps : *s
    *cond ? *ps : *ps    
  END
