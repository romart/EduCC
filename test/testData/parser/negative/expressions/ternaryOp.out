test/testData/parser/negative/expressions/ternaryOp.c:10:15: warning: pointer/integer type mismatch in conditional expression ('signed int' and '*signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:16: warning: pointer/integer type mismatch in conditional expression ('signed int' and '*float')
test/testData/parser/negative/expressions/ternaryOp.c:10:17: warning: pointer/integer type mismatch in conditional expression ('signed int' and 'signed int[20]')
test/testData/parser/negative/expressions/ternaryOp.c:10:19: warning: pointer/integer type mismatch in conditional expression ('signed int' and '*STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:27: warning: pointer/integer type mismatch in conditional expression ('*signed int' and 'signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:30: warning: pointer type mismatch ('*signed int' and '*float')
test/testData/parser/negative/expressions/ternaryOp.c:10:33: warning: pointer type mismatch ('*signed int' and '*STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:34: warning: pointer/integer type mismatch in conditional expression ('*float' and 'signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:36: warning: pointer type mismatch ('*float' and '*signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:38: warning: pointer type mismatch ('*float' and 'signed int[20]')
test/testData/parser/negative/expressions/ternaryOp.c:10:40: warning: pointer type mismatch ('*float' and '*STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:41: warning: pointer/integer type mismatch in conditional expression ('signed int[20]' and 'signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:44: warning: pointer type mismatch ('signed int[20]' and '*float')
test/testData/parser/negative/expressions/ternaryOp.c:10:47: warning: pointer type mismatch ('signed int[20]' and '*STRUCT S')
test/testData/parser/negative/expressions/ternaryOp.c:10:55: warning: pointer/integer type mismatch in conditional expression ('*STRUCT S' and 'signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:57: warning: pointer type mismatch ('*STRUCT S' and '*signed int')
test/testData/parser/negative/expressions/ternaryOp.c:10:58: warning: pointer type mismatch ('*STRUCT S' and '*float')
test/testData/parser/negative/expressions/ternaryOp.c:10:59: warning: pointer type mismatch ('*STRUCT S' and 'signed int[20]')
